common/src/ui/BoxSelectionTool.h

```diff
@@ -0,0 +1,76 @@
+//Added by Lws
+
+#pragma once
+
+#include "ui/HandleDragTracker.h"
+#include "vm/bbox.h"
+
+#include <memory>
+
+namespace tb::render
+{
+class RenderContext;
+class RenderBatch;
+}
+
+namespace tb::mdl
+{
+class Node;
+}
+
+namespace tb::ui
+{
+class DrawShapeTool;
+class MapDocument;
+class InputState;
+struct DragState;
+
+// Box selection renderer - used to display the box selection area
+class SelectionBoxRenderer
+{
+private:
+  vm::bbox3d m_selectionBounds;
```

Comment:
Why duplicate the state here? I think the rendering code could just be extracted into free functions and called directly from the delegate which has the state anyway.